docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+When this setting is set to `true`, multiple connections are allowed from the same IP address (for example, on different
+ports).
+
+### p2p.handshake_timeout
+
+Timeout duration for protocol handshake (or secret connection negotiation).
+
+```toml
+handshake_timeout = "20s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+This high-level timeout value is applied when the TCP connection has been
+established with a peer, and the node and peer are negotiating its upgrade into
+a secret authenticated connection.
+
+The value `"0s"` is undefined, and it can lead to unexpected behaviour.
+
+### p2p.dial_timeout
+
+Timeout duration for the low-level dialer that connects to the remote address on the TCP network.
+
+```toml
+dial_timeout = "3s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+This parameter is the timeout value for dialing on TCP networks. If a hostname is used instead of an IP address and the
+hostname resolves to multiple IP addresses, the timeout is spread over each consecutive dial, such that each is given an
+appropriate fraction of the time to connect.
+
+Setting the value to `"0s"` disables the timeout.
+
+## Mempool
+Mempool allows gathering and broadcasting uncommitted transactions among nodes.
+
+The **mempool** is storage for uncommitted transactions; the **mempool cache** is internal storage within the
+mempool for seen transactions. The mempool cache provides a list of transactions already received to filter out
+incoming duplicate transactions and prevent duplicate full transaction validations.
+
+### mempool.type
+The type of mempool this node will use.
+```toml
+type = "flood"
+```
+
+| Value type          | string    |
+|:--------------------|:----------|
+| **Possible values** | `"flood"` |
+|                     | `"nop"`   |
+
+`"flood"` is the original mempool implemented for CometBFT. It is a concurrent linked list with flooding gossip
+protocol.
+
+`"nop"` is a "no operation" or disabled mempool, where the ABCI application is responsible for storing, disseminating and
+proposing transactions. Note, that it requires empty blocks to be created:
+[`consensus.create_empty_blocks = true`](#consensuscreate_empty_blocks) has to be set.
+
+### mempool.recheck
+Transaction validity check.
+```toml
+recheck = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+Committing a block affects the application state, hence the remaining transactions in the mempool after a block commit
+might become invalid. Setting `recheck = true` will go through the remaining transactions and remove invalid ones.
+
```

Comment:
```suggestion

If you application may remove transactions in its `PrepareProposal` handler,
you probably need to set this configuration to `true` to avoid possible leaks in your mempool
(transactions staying in the mempool until the node is next restarted).
```